### PR TITLE
Payam/dicom manipulation cpp code

### PR DESCRIPTION
**Pull request by PayamZandiyeh**
@Besler @xtinecow 
DRR.py is working but I am not sure what is happening with Read/Write function of itk. If I read my 3D image and write it without doing anything the 3D image will be shifted along the z-direction. Also, the direction of the image will be changed. 
Otherwise, the DRR.py is working properly. I have outlined the issue in the Issues to be addressed by myself later. 
I am merging this branch since I need the Classes written by @xtinecow to continue my DRR generation code. 
@Besler Please advise 
Thanks PZ

### CODE REVIEW

File: Registration/DRR Generation/Python/DRR.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# This script generates Digitally Reconstructed Radiograph (DRR) from a given 
+# 3D image to given 2D image plane(s)
+
+# ISSUES - 
+# 1- I don't know how to set the direction of the output image
+
+import itk # imports insight Toolkit
+import numpy as np
+import sys
+from Functions import ChangeImageDirection
```
Comment (Besler) [inline Registration/DRR Generation/Python/DRR.py:10]: This is fine, but in general you should avoid `from A import B`. Reduces the maintainability of your code. This is especially true for large files.

File: Registration/DRR Generation/Python/DRR.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+
+#%% -------------------------------- Generated Inputs --------------------------------- 
+direction = np.identity(3,dtype=np.double)
+#direction[0,:] = [np.sqrt()/2,-np.sqrt(1)/2,0.] # Creating a non identity direction output just for testing. 
+#direction[1,:] = [np.sqrt(1)/2, np.sqrt(3)/2,0.] # Creating a non identity direction output just for testing. 
+
+
+#%% -------------------------------- INPUTS --------------------------------- 
+input_filename  = '/Volumes/Storage/Projects/Registration/QuickData/OA-BEADS-CT.nii'
```
Comment (Besler) [inline Registration/DRR Generation/Python/DRR.py:21]: It's ususally not good to hard code your input file names. It is preferable that you set them as command line arguments. I can clean this up later though.

File: Registration/DRR Generation/Python/Functions.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        for j in range(DimensionOut):
+            vnlMatrix.put(i,j,newDirection[i][j])
+
+
+        
+def PrintDirection(imageDirection,DimensionOut):
+    vnlMatrix = imageDirection.GetVnlMatrix()
+    for i in range(DimensionOut):
+        for j in range(DimensionOut):
+            print "{:>8.4f}".format(vnlMatrix.get(i,j)),
```
Comment (Besler) [inline Registration/DRR Generation/Python/Functions.py:22]: If you use the `print` as a function call (`print('hello world')` your code will be python 3 compatible.

**Comment by Besler** [comment]
Looking good @PayamZandiyeh! Once merged with master I can make it part of the cli.

File: Registration/DRR Generation/Python/DRR.py (hunk 2 above)
Comment (PayamZandiyeh) [reply Registration/DRR Generation/Python/DRR.py:21]: @Besler I agree ... I am going to change it today by using xtine's Classes.

File: Registration/DRR Generation/Python/Functions.py (hunk 3 above)
Comment (PayamZandiyeh) [reply Registration/DRR Generation/Python/Functions.py:22]: @Besler OK will do so.

File: Registration/DRR Generation/Python/DRR.py (hunk 1 above)
Comment (PayamZandiyeh) [reply Registration/DRR Generation/Python/DRR.py:10]: @Besler  What do you suggest instead? If I want to get some defs from a class per say in the current folder or elsewhere, how can I do so by avoiding "from"? 
Thanks

Comment (Besler) [reply Registration/DRR Generation/Python/DRR.py:10]: You can do something like
```python
# Imports
import Functions

# After much code...
Functions. ChangeImageDirection(image)
```

It's all good for now though.

File: Registration/DRR Generation/Python/DRR.py (hunk 2 above)
Comment (Besler) [reply Registration/DRR Generation/Python/DRR.py:21]: Okay, awesome!

File: Registration/DRR Generation/Python/DRR.py (hunk 1 above)
Comment (PayamZandiyeh) [reply Registration/DRR Generation/Python/DRR.py:10]: I can do it on my functions that are in my own directory. However, if I want to import the class the Christine has written in her folder, it doesn't work. 
But i get your point 👍

**Pull request merged**